scripts/ai/AIDriveStrategyUnloadCombine.lua

```diff
@@ -619,24 +652,37 @@ function AIDriveStrategyUnloadCombine:getCourseToAlignTo(vehicle, offset)
     return tempCourse
 end
 
-function AIDriveStrategyUnloadCombine:getTrailersTargetNode()
-    local trailer = AIUtil.getImplementOrVehicleWithSpecialization(self.vehicle, Trailer)
-    if trailer then
-        if self.combineToUnload:getCpDriveStrategy():canLoadTrailer(trailer) then
-            local targetNode = trailer:getFillUnitAutoAimTargetNode(1)
-            if targetNode then
-                return targetNode
-            else
-                self:debugSparse('Can\'t find trailer target node')
+--- Gets fill nodes of all fill units.
+---@return table|nil
+function AIDriveStrategyUnloadCombine:getTrailersTargetNodes()
+    local targetNodes = {}
+    local trailers = AIUtil.getAllChildVehiclesWithSpecialization(self.vehicle, Trailer)
+    if not trailers then 
+        self:debugSparse('Can\'t find any trailers')
+        return
+    end
+    for _, trailer in pairs(trailers) do 
+        for ix, _ in pairs(trailer:getFillUnits()) do 
+            local node = trailer:getFillUnitExactFillRootNode(ix)
+            local _, _, trailerOffset = localToLocal(node, trailer.rootNode, 0, 0, 0)
+            local _, _, vehicleOffset = localToLocal(trailer.rootNode, self.vehicle.rootNode, 0, 0, 0)
+            if node then 
+                table.insert(targetNodes, {
+                    node = node,
+                    fillUnitIx = ix,
+                    trailer = trailer,
+                    trailerOffset = trailerOffset,
+                    vehicleOffset = vehicleOffset,
+                })
             end
-        else
-            self:debugSparse('Combine says it can\'t load trailer')
-            --TODO: maybe then send the unloader away if activated?
         end
-    else
-        self:debugSparse('Can\'t find trailer')
     end
-    return trailer.rootNode
+    table.sort(targetNodes, function (a, b)
+        --- Sorts these nodes to make sure the front most node is the first.
```

Comment:
I remember another issue, just can't find it, with overloading, where this sorting could be useful, maybe consider moving this into the SelfUnloadHelper so combines can also use it?